### PR TITLE
[native] Add config parameters for remote function execution

### PR DESCRIPTION
**Pull request by pedroerp**
## Description
Adding more configuration parameters in native executor to support remote function server configuration. The following parameters were added:
* remote-function-server.thrift.address: ip or dns address of a remote function server (default to loopback address)
* remote-function-server.thrift.port: its port
* remote-function-server.thrift.uds-path: alternatively, specify a unix domain socket file path for communication (if local)
* remote-function-server.signature.files.directory.path: path containing json signature function files.

## Test Plan
Unit tests added. The next PR will include a framework and end-to-end tests. 

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by mbasmanova — APPROVED** [review]
@pedroerp It would be nice to update commit message to list all the new configs and explain how to use these. It would be great to start documenting configs like in https://facebookincubator.github.io/velox/configs.html or https://prestodb.io/docs/current/admin/properties.html

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 1]
```diff
@@ -845,3 +845,53 @@ The following properties allow tuning the :doc:`/functions/regexp`.
     to hit the limit on matches for subsequent rows as well, you want to use the
     correct algorithm from the beginning so as not to waste time and resources.
     The more rows you are processing, the larger this value should be.
+
+Remote Function Execution
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/admin/properties.rst:849]: Thank you for documenting new configs.

**Review by pranjalssh — APPROVED** [review]

**Comment by tdcmeehan** [comment]
Can this incorporate more high level information, and be reconciled with [existing documentation]https://prestodb.io/docs/current/admin/function-namespace-managers)?

**Comment by pedroerp** [comment]
> Can this incorporate more high level information

I'm planning to write a document about the high-level architecture when the last few bits are hashed out. My plan is to have a Velox-specific page (since the remote function support is a Velox feature), then I could put a Presto doc together on how this is integrated in Presto, the config parameters, etc.

This PR only adds the config parameters missing, but there's still some more plumbing to be done before this is entirely supported. I'm keeping track of the remaining items in this Issue:

https://github.com/facebookincubator/velox/issues/5288

> and be reconciled with [existing documentation](https://prestodb.io/docs/current/admin/function-namespace-managers)? Perhaps by broadening the section into general user-defined functions support?

This specific document is, in a way, orthogonal. With the current design, any function can be registered as remote in Velox (regardless of how they get registered in Presto). This page also seems to be missing documentation about the json registration method we are currently planning on using, but unfortunately I don't really have enough context to write a doc about it myself. 

Maybe @rschlussel or @Ali-P could help?

**Comment by pedroerp** [comment]
@tdcmeehan let me know if you have any concerns with this approach. Otherwise I'll go ahead and merge this PR (and will tag you once I submit the PR with more docs).

**Comment by pedroerp** [comment]
@tdcmeehan and I discussed this offline. The plan is to remove these configs from the administration documentation as the latter are meant to be end-user facing. 

I'll submit a new PR and add a new section for Prestissimo under "Developer Guide" and start adding the remote function support instructions there.

**Pull request merged**